vumi_http_proxy/test/test_http_proxy.py

```diff
@@ -1,30 +1,28 @@
 from twisted.internet import reactor
-from twisted.internet.defer import inlineCallbacks, Deferred, returnValue
-from twisted.internet.endpoints import serverFromString, clientFromString
-from twisted.trial import unittest
-from twisted.test import proto_helpers
+from twisted.internet.defer import inlineCallbacks, returnValue
+from twisted.internet.endpoints import clientFromString, serverFromString
 from twisted.web.client import ProxyAgent, readBody
-
+from twisted.trial import unittest
 from vumi_http_proxy.http_proxy import ProxyFactory
-
-from .helpers import HttpTestServer, DEFAULT_TIMEOUT
+from vumi_http_proxy.test import helpers
 
 
-class TestProxyToLocalServer(unittest.TestCase):
-
-    timeout = DEFAULT_TIMEOUT
+class TestCheckProxyRequest(unittest.TestCase):
+    timeout = helpers.DEFAULT_TIMEOUT
 
     def setUp(self):
-        self.server = HttpTestServer(self)
+        self.server = helpers.HttpTestServer(self)
 
     @inlineCallbacks
     def setup_proxy(self, blacklist):
-        proxy = ProxyFactory(blacklist)
+        resolver = helpers.TestResolver()
+        http_client = helpers.TestAgent()
+        proxy = ProxyFactory(blacklist, resolver, http_client)
         server_endpoint = serverFromString(
-            reactor, "tcp:0:interface=127.0.0.1")
-        server = yield server_endpoint.listen(proxy)
-        self.addCleanup(server.stopListening)
-        returnValue(server.getHost().port)
+            reactor, "tcp:8080:interface=127.0.0.1")
```

Comment:
I think we want to replace `8080` here with `0`. `0` should hopefully cause the server to listen on a arbitrary open port which we will then return at the end of the function (see the line with `self.server.getHost().port`).